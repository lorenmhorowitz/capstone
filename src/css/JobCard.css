.card {
  max-width: 25%;
}

.cardMedia {
  height: 100%;
  width: 100%;
  object-fit: cover;
}
