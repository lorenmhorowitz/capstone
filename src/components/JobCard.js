import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import "../css/JobCard.css";

const styles = {
  card: {
    maxWidth: 300
  },
  media: {
    objectFit: "cover"
  }
};

class ImgMediaCard extends Component {
  render() {
    return (
      <Card className="card">
        <CardActionArea>
          <CardMedia
            component="img"
            alt="Nice house"
            className="cardMedia"
            image={this.props.image}
            title="Nice house"
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              House
            </Typography>
            <Typography component="p">You like this house</Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    );
  }
}

ImgMediaCard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(ImgMediaCard);
