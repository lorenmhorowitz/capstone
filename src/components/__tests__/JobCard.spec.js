import React from "react";
import JobCard from "../JobCard";
import { shallow } from "enzyme";

describe("AppBar Component", () => {
  it("JobCard renders correctly", () => {
    const image =
      "https://upload.wikimedia.org/wikipedia/commons/2/25/Houses_on_Garfield_Place,_Poughkeepsie,_NY.jpg";
    let wrapper = shallow(<JobCard image={image} />);
    expect(wrapper).toMatchSnapshot();
  });
});
