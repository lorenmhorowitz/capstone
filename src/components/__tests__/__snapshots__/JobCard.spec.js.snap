// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component JobCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ImgMediaCard)
    image="https://upload.wikimedia.org/wikipedia/commons/2/25/Houses_on_Garfield_Place,_Poughkeepsie,_NY.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "card": "ImgMediaCard-card-1",
        "media": "ImgMediaCard-media-2",
      },
      "image": "https://upload.wikimedia.org/wikipedia/commons/2/25/Houses_on_Garfield_Place,_Poughkeepsie,_NY.jpg",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "card": "ImgMediaCard-card-1",
          "media": "ImgMediaCard-media-2",
        },
        "image": "https://upload.wikimedia.org/wikipedia/commons/2/25/Houses_on_Garfield_Place,_Poughkeepsie,_NY.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
